lib/resources/postgres.rb

```diff
@@ -81,6 +60,32 @@ def verify_dirs
       end
     end
 
+    def version_from_psql
+      return unless inspec.command('psql').exist?
+      inspec.command("psql --version | awk '{ print $NF }' | awk -F. '{ print $1\".\"$2 }'").stdout.strip
+    end
+
+    def verify_data_dir_location(ver = @version)
```

Comment:
the @version instance var is not modified it is just used in the loop of the dir list, this function now just returns the instance var is it supporting rather than modifying it directly.